bee-node/build.rs

```diff
@@ -36,3 +79,89 @@ fn main() -> Result<(), BuildError> {
 
     Ok(())
 }
+
+#[cfg(feature = "dashboard")]
+mod dashboard {
+    use super::*;
+
+    use serde::Deserialize;
+    use zip::ZipArchive;
+
+    use std::{
+        io::{self, BufReader, Cursor},
+        path::Path,
+    };
+
+    const ASSET_PREFIX: &str = "node-dashboard-bee-";
+    const RELEASES_URL: &str = "https://api.github.com/repos/iotaledger/node-dashboard/releases/latest";
+
+    #[derive(Deserialize, Debug)]
+    struct LatestReleaseAssets {
+        assets: Vec<ReleaseAsset>,
+    }
+
+    #[derive(Deserialize, Debug)]
+    struct ReleaseAsset {
+        name: String,
+        browser_download_url: String,
+    }
+
+    pub(super) async fn fetch<P: AsRef<Path>>(dashboard_dir: P) -> Result<(), BuildError> {
+        println!("fetching latest dashboard release");
+
+        let client = reqwest::Client::builder()
+            .user_agent("bee-fetch-dashboard")
+            .build()
+            .expect("could not create client");
+
+        let response = client
+            .get(RELEASES_URL)
+            .send()
+            .await
+            .and_then(|resp| resp.error_for_status())
+            .map_err(|e| {
+                BuildError::DashboardRequest(e.status().map(|code| code.as_u16()), RELEASES_URL.to_string())
+            })?;
+
+        let assets = response
+            .json::<LatestReleaseAssets>()
+            .await
+            .map_err(|_| BuildError::DashboardDecode)?
+            .assets;
+
+        let release_asset = assets
+            .iter()
+            .find(|asset| asset.name.starts_with(ASSET_PREFIX))
+            .ok_or(BuildError::DashboardNoSuitableRelease)?;
+
+        println!(
+            "downloading latest dashboard release `{}` from {}",
+            release_asset.name, release_asset.browser_download_url
+        );
+
+        let mut archive = tempfile::tempfile().expect("could not create temp file");
+
+        let response = client
+            .get(&release_asset.browser_download_url)
+            .send()
+            .await
+            .and_then(|resp| resp.error_for_status())
+            .map_err(|e| {
+                BuildError::DashboardRequest(
+                    e.status().map(|code| code.as_u16()),
+                    release_asset.browser_download_url.clone(),
+                )
+            })?;
+
+        let mut content = Cursor::new(response.bytes().await.map_err(|_| BuildError::DashboardDownload)?);
+        io::copy(&mut content, &mut archive).expect("copying failed");
+        let mut archive = ZipArchive::new(BufReader::new(archive)).map_err(|_| BuildError::DashboardInvalidArchive)?;
+
```

Comment:
We should also check the SHA checksum to make sure the download was not corrupted or MITM'ed.